src/create-glamorous.js

```diff
@@ -42,26 +45,29 @@ function createGlamorous(splitProps) {
        */
       const GlamorousComponent = withTheme(
         (props, context) => {
-          /* eslint no-use-before-define: 0 */
+          const updateClassName = shouldUpdate(props, context)
+
           const {toForward, cssOverrides, cssProp} = splitProps(
             props,
             GlamorousComponent,
           )
 
           // create className to apply
-          const className = getGlamorClassName({
-            styles: GlamorousComponent.styles,
-            props,
-            cssOverrides,
-            cssProp,
-            context,
-            displayName: GlamorousComponent.displayName,
-          })
+          GlamorousComponent.className = updateClassName ?
```

Comment:
Note that these changes are only in `next` right now, so if you make a PR, make sure you base your branch on `next`.